frontend/src/scenes/notebooks/Nodes/NotebookNodeInsight.tsx

```diff
@@ -31,11 +34,25 @@ export const NotebookNodeInsight = createPostHogWidgetNode<NotebookNodeInsightAt
     startExpanded: true,
     attributes: {
         id: {},
+        title: {},
     },
     pasteOptions: {
         find: urls.insightView('(.+)' as InsightShortId),
         getAttributes: async (match) => {
-            return { id: match[1] as InsightShortId }
+            const shortId = match[1] as InsightShortId
```

Comment:
If we do the above we could move this to the logic or into a `useEffect` hook on the `NodeWrapper`. Something like:

```
useEffect(() => {
  if (!node.attrs.title) {
      const title = typeof defaultTitle === "function" ? defaultTitle(attributes) : defaultTitle
      updateAttributes({ title: title })
  }
}, [])

// then in the component you can have either

title: 'Random Node with no customization'
title: (attributes) => {
 // logic to fetch title from api
}
```